lib/InteractiveViewer.ml

```diff
@@ -0,0 +1,120 @@
+open Nottui
+module W = Nottui_widgets
+open Lwd_infix
+
+type view_mode = SideBySide | Normal
+
+let view_mode : view_mode Lwd.var = Lwd.var Normal
+
+let toggle_view_mode () : unit =
+  match Lwd.peek view_mode with
+  | Normal -> Lwd.set view_mode SideBySide
+  | SideBySide -> Lwd.set view_mode Normal
+
+let help_visible = Lwd.var false
+let quit = Lwd.var false
+
+let toggle_help_visibility () =
+  Lwd.set help_visible (not (Lwd.peek help_visible))
+
+let view (patches : Patch.t list) =
+  let z_patches_var : Patch.t Zipper.t Lwd.var =
+    match Zipper.zipper_of_list patches with
+    | Some z -> Lwd.var z
+    | None -> failwith "zipper_of_list: empty list"
+  in
+  let hunks_ui =
+    let$ mode = Lwd.get view_mode and$ z_patches = Lwd.get z_patches_var in
+    match mode with
+    | Normal -> HunkView.current_hunks z_patches
+    | SideBySide -> HunkView.current_hunks_side_by_side z_patches
+  in
+  let curr_scroll_state = Lwd.var W.default_scroll_state in
+  let change_scroll_state _action state =
+    let off_screen = state.W.position > state.W.bound in
+    if off_screen then
+      Lwd.set curr_scroll_state { state with position = state.W.bound }
+    else Lwd.set curr_scroll_state state
+  in
+  let ui =
+    let$ help_visible = Lwd.get help_visible in
+    if help_visible then
+      W.vbox
+        [
+          W.scrollbox @@ Lwd.pure @@ HelpView.help_panel;
+          Lwd.pure
+          @@ Ui.keyboard_area
+               (function
+                 | `ASCII 'q', [] ->
+                     toggle_help_visibility ();
+                     `Handled
+                 | _ -> `Unhandled)
+               (W.string "Type 'q' to exit the help panel");
+        ]
+    else
+      W.vbox
+        [
+          Lwd.pure (OperationView.operation_info (Lwd.peek z_patches_var));
```

Comment:
There is a bug where the changes are not reactive.. I think the solution lies somewhere in Lwd.map but after trying I can't seem to figure it out. Can I ask your advice on what I should do to resolve this? Thanks